backend/backendserver/src/main/java/com/github/klucsik/birdnoiseserver/backendserver/services/DevicePlayParamSelectorService.java

```diff
@@ -43,17 +44,34 @@ public DevicePlayParamSlimDto selectSlimPlayParam(String chipId, Long paramVersi
         switch (playParamsForDevice.size()) {
             case 1:
                 if (paramVersion == playParamsForDevice.get(0).getPlayParam().getId()) {
-                    return null;
+                    //current playPAram is on the device
+                    if (playParamsForDevice.get(0).getStatus() == DPPStatus.DEPLOYING) {
+                        //successfully deployed change status
+                        playParamsForDevice.get(0).setStatus(DPPStatus.DEPLOYED);
+                        return null; //because its the same on the device and on the server
+                    }
+                    else {
+                        playParamsForDevice.get(0).setStatus(DPPStatus.DEPLOYED); //Because the playParam on the device is good
+                        return null;                                              //TODO: but send a warn! log
+                    }
                 }
                 else {
-                    PlayParamDto dto = PlayParamMapper.MAPPER.playParamtoDto(playParamsForDevice.get(0).getPlayParam());
-                    return DevicePlayParamSlimMapper.MAPPER.playParamDtotoDevice(dto);
+                    PlayParam oldPlayParam = playParamRepository.getOne(paramVersion);
```

Comment:
The device will send bullshit in paramVersion when it first boots up. handle the nullpointer in the next line pls